### PR TITLE
feat: allow edit of commonJs plugin params

### PR DESCRIPTION
**Pull request by huang-julien**
- add commonjs field to edit the commonjs plugin

### 🔗 Linked issue
https://github.com/nuxt/framework/pull/4762
https://github.com/nuxt/framework/issues/4661

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add the commonJs key to NitroOptions and assign by destructuring nitro.option.commonJs  when adding the @rollup/plugin-commonjs.
Would improve the linked PR by avoiding replacing the commonJS plugin at build. The commonJS dynamicRequireTargets declaration would be directly into nitro/nuxt config

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/types/nitro.ts [hunk 1]
```diff
@@ -139,6 +140,7 @@ export interface NitroOptions {
   externals: NodeExternalsOptions
   analyze: false | PluginVisualizerOptions
   replace: Record<string, string | ((id: string) => string)>
+  commonJs?: RollupCommonJSOptions
```
Comment (pi0) [inline src/types/nitro.ts:143]: Can you please use `commonJS` for naming?

**Review by pi0 — COMMENTED** [review]
Thanks! Would be nice to also update docs (config reference to mention this new option)

File: src/types/nitro.ts (hunk 1 above)
Comment (huang-julien) [reply src/types/nitro.ts]: done \:+1:

File: docs/config/index.md [hunk 2]
```diff
@@ -297,3 +297,6 @@ Rollup specific option. Specifies module imports that have side-effects
 
 Rollup specific option.
 
+### `commonJS`
+
+Rollup specific option. Specifies additionnal configuration for the rollup commonJS plugin.
```
Comment (danielroe) [inline docs/config/index.md:302]:
```suggestion
Rollup specific option. Specifies additional configuration for the rollup CommonJS plugin.
```

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**